### PR TITLE
[chore] docs(release): add merging ready PRs

### PR DESCRIPTION
**Pull request by astencel-sumo**
Changes:

- https://github.com/open-telemetry/opentelemetry-collector/commit/fdaebfdddc6e8c9d363823d6d0fb458c0585e5de adds a point on merging ready pull requests to releasing Core and Contrib
- https://github.com/open-telemetry/opentelemetry-collector/commit/ed8db0229d77151fb7b9dec83af90562e17fa041 "fixes" numbering of the two modified ordered lists using static numbering `1.`, `1.`, `1.`. Not sure if this is a controversial change. I can change it to regular dynamic points `1.`, `2.`, `3.`... if required.

Update:

- https://github.com/open-telemetry/opentelemetry-collector/pull/9611/commits/08348d7805d8386d12e481920935c4bee7ac75a5 reverts https://github.com/open-telemetry/opentelemetry-collector/commit/ed8db0229d77151fb7b9dec83af90562e17fa041 on [request](https://github.com/open-telemetry/opentelemetry-collector/pull/9611#discussion_r1496276335) from Tyler.

### CODE REVIEW

File: docs/release.md
Comment (TylerHelmuth) [inline docs/release.md:28]: I believe these steps are supposed to be ordered, so doing `1 -> 2 -> 3` makes more sense to me.

Comment (astencel-sumo) [reply docs/release.md:28]: Reverted the numbering change.

**Review by TylerHelmuth — APPROVED** [review]

**Review by jpkrohling — APPROVED** [review]

**Comment by jpkrohling** [comment]
How about we have a schedule where the release manager checks on Thu EOD (EMEA) or Fri morning (NASA) for such items and ask maintainers to merge them? 

In any case, https://github.com/open-telemetry/opentelemetry-collector-releases/pull/476 made me realize that we are also missing a step to check for release blockers.

**Comment by djaglowski** [comment]
> PRs staying in ready to merge for days and even weeks. Not sure how else this issue can be resolved.

I think the problem is likely that no maintainers are aware that these issues are ready to merge. Perhaps we could add an automated check to periodically @ the maintainers. We could also ping them immediately when the label is applied, but often the reason the label is applied is because merging is blocked for some reason. e.g. a release is being prepared.

**Comment by dmitryax** [comment]
I see the problem that `ready to merge` PRs sometimes take longer to merge, but I don't think we should bring this to the release procedure. The release takes a significant of time already and often involves going the extra mile to ensure the build is green before the release. This extra step makes it more complicated. 

Also, sometimes, the `ready to merge` label is set on PRs that would benefit from another set of eyes. 

I think solving this problem should be detached from the release procedure.

**Comment by astencel-sumo** [comment]
Closing this PR then.

**Pull request closed without merging**
